src/Microsoft.Azure.Mobile.Client/Platforms/ios/Push/Push.cs

```diff
@@ -107,8 +108,17 @@ internal static string ParseDeviceToken(NSData deviceToken)
             {
                 throw new ArgumentNullException("deviceToken");
             }
-
-            return deviceToken.Description.Trim('<', '>').Replace(" ", string.Empty).ToUpperInvariant();
+            byte[] byteArray = deviceToken.ToArray();
+            if (byteArray.Length == 0)
+            {
+                throw new ArgumentException("deviceToken.ToArray() returned empty array.");
```

Comment:
Let's rephrase it so it is easier to understand.

```suggestion
                throw new ArgumentException("deviceToken cannot be empty.");
```